Instruments/bridge12.py

```diff
@@ -404,25 +380,46 @@ def grab_consist_power():
         self.tuning_curve_data[sweep_name+'_freq'] = freq
         self.last_sweep_name = sweep_name
         return rxvalues, txvalues
+    def handle_midpoint_failure(self):
+        result = input("Couldn't fine the midpoint; maybe the wg didn't turn on completely, try again?")
+        if result.lower().startswith("y"):
+            wg_engaged = False
+        else:
+            self.set_rf(False)
+            self.set_wg(False)
+            raise ValueError("The reflection of the first point is the same or lower than the rx of the dip, which doesn't make sense -- check %gdBm_%s"%(10.0,'rx'))
     def lock_on_dip(self, ini_range=(9.81e9,9.83e9),
             ini_step=0.5e6,# should be half 3 dB width for Q=10,000
             dBm_increment=3, n_freq_steps=15):
         """Locks onto the main dip, and finds the first polynomial fit also sets the current frequency bounds."""    
         if not self.frq_sweep_10dBm_has_been_run:
-            logger.info("Did not find previous 10 dBm run, running now")
-            self.set_wg(True)
-            self.set_amp(True)
-            self.set_rf(True)
-            self.set_power(10.0)
             freq = r_[ini_range[0]:ini_range[1]:ini_step]
-            rx, tx = self.freq_sweep(freq)
+            logger.info("ini range: "+str(ini_range)+"ini step: "+str(ini_step))
+            wg_engaged = False
+            logger.info("Did not find previous 10 dBm run, running now")
+            while not wg_engaged:
+                try:
+                    self.set_wg(True)
+                    self.set_rf(True)
+                    self.set_amp(True)
+                    self.set_power(10.0)
+                    rx, tx = self.freq_sweep(freq)
+                    rx,freq = [self.tuning_curve_data['%gdBm_%s'%(10.0,j)] for j in ['rx','freq']]
+                    rx_dBm = convert_to_power(rx)
+                    rx_midpoint = (max(rx_dBm) + min(rx_dBm))/2.0
+                    #is the first rx higher than the midpoint (rx of dip/2)? If not this means we are not looking at a dip - ex. if the wg didn't switch on our rx would be a straight line and we would not have a dip
+                    if not (rx_dBm > rx_midpoint)[0]:
+                        self.handle_midpoint_failure()
+                    else:
+                        wg_engaged = True
+                except:
+                    self.handle_midpoint_failure()
+        else:
+            rx,freq = [self.tuning_curve_data['%gdBm_%s'%(10.0,j)] for j in ['rx','freq']]
+            rx_dBm = convert_to_power(rx)
+            rx_midpoint = (max(rx_dBm) + min(rx_dBm))/2.0
         assert self.frq_sweep_10dBm_has_been_run, "I should have run the 10 dBm curve -- not sure what happened"
-        rx,freq = [self.tuning_curve_data['%gdBm_%s'%(10.0,j)] for j in ['rx','freq']]
-        rx_dBm = convert_to_power(rx)
-        rx_midpoint = (max(rx_dBm) + min(rx_dBm))/2.0
```

Comment:
you need this...

Reply:
again it is up in the try block

Reply:
again, re-read the last comment

Reply:
again it's in the else block ^mispoke above